test/cmocka/include/mock_trace.h

```diff
@@ -0,0 +1,37 @@
+#include <sof/preproc.h>
+
+#define _TRACE_EVENT_NTH_PARAMS(id_count, param_count)			\
+	uintptr_t log_entry						\
+	META_SEQ_FROM_0_TO(id_count   , META_SEQ_STEP_id_uint32_t)	\
+	META_SEQ_FROM_0_TO(param_count, META_SEQ_STEP_param_uint32_t)
+
+#define _TRACE_EVENT_NTH(postfix, param_count)			\
+	META_FUNC_WITH_VARARGS(					\
+		_trace_event, META_CONCAT(postfix, param_count),\
+		void, _TRACE_EVENT_NTH_PARAMS(2, param_count)	\
+	)
```

Comment:
Couldn't we reuse existing `_TRACE_EVENT_NTH`?
I understand that implementation must be different, but the declaration?

Reply:
No need to worry about UTs being future-proof atm, probably I'll change test framework after Kconfig, because current one has too many issues, so these changes won't matter anyway.